cabal-install/src/Distribution/Client/ProjectConfig.hs

```diff
@@ -559,18 +559,17 @@ readProjectConfig :: Verbosity
                   -> Flag FilePath
                   -> DistDirLayout
                   -> Rebuild ProjectConfigSkeleton
-readProjectConfig verbosity httpTransport ignoreProjectFlag configFileFlag distDirLayout = do
-    global <- singletonProjectConfigSkeleton <$> readGlobalConfig verbosity configFileFlag
-    local  <- readProjectLocalConfigOrDefault verbosity httpTransport distDirLayout
-    freeze <- readProjectLocalFreezeConfig    verbosity httpTransport distDirLayout
-    extra  <- readProjectLocalExtraConfig     verbosity httpTransport distDirLayout
-    if ignoreProjectFlag == Flag True then return (global <> (singletonProjectConfigSkeleton defaultProject))
-    else return (global <> local <> freeze <> extra)
-    where
-      defaultProject :: ProjectConfig
-      defaultProject = mempty {
-        projectPackages = ["./"]
-      }
+readProjectConfig verbosity httpTransport ignoreProjectFlag configFileFlag distDirLayout =
+  if ignoreProjectFlag == Flag True
```

Comment:
Maybe this could be done with a guard instead of and `if`? It's a stylistic wibble, of course, so please feel free to leave it.

Reply:
Sure 👍